static/js/datatables_config.js

```diff
@@ -0,0 +1,103 @@
+const gen_datatables_default_config = (table_selector) => {
+  const column_selector = (idx, data, node) => {
+	// https://datatables.net/forums/discussion/42192/exporting-data-with-buttons-and-responsive-extensions-controlled-by-column-visibility
+	return $(table_selector).DataTable().column( idx ).visible();
+  };
+  return {
+	dom: 'Bfrtipl',
+	responsive: true,
+	colReorder: true,
+	deferRender: true,
+	createdRow: (row) => {
+	  $(row).find('[data-toggle="popover"]').popover();
+	},
+	columnDefs: [
+	  {
+		targets: 'datatables-points',
+		render: (data, type, row) => {
+			if(type === 'display') {
+			  return data;
+			} else {
+				const div = document.createElement("div");
+				div.innerHTML = data;
+				const percent = div.innerText;
+			  if(type === 'export') {
+				return percent;
+			  } else {
+				return parseFloat(percent.replace('%', ''));
+			  }
+			}
+		  }
+	  },
+	  {
+		targets: 'datatables-status',
+		render: (data, type, row) => {
+		  const isExport = type === 'export';
+		  if(isExport || type === 'display') {
+			if(data == 'Z')
+			  return isExport ? 'Zakwalifikowany' : '<span class="qualified">Zakwalifikowany</span>';
+			else if(data == "None")
+			  return '';
+			else if(data == 'O')
+			  return isExport ? 'Odrzucony' : '<span class="not-qualified">Odrzucony</span>' ;
+			else
+			  return data;
+		  } else
+			return data;
+		}
+	  },
+	  {
+		targets: 'datatables-yes-no',
+		render: (data, type, row) => {
+		  if(type === 'export') {
+			return data.split("> ").pop();
```

Comment:
Can't we just, like, remove all HTML tags rather than doing weird things depending on the column?

Also I still want the checkmark in PDF and print views :disappointed:

Reply:
Spent too much time today trying to get checkmarks working in the pdf. Changing the font in pdfmake to one which does support the checkmark character doesn't work.

I won't fix the checkmarks as this is not important right now and might be done later.

Reply:
Okay, I guess we can leave the checkmarks for now. But what about the first question - do we need to differentiate columns by type and do custom handling, or would stripping all HTML for exporting be enough? (except for removing the checkmark as already discussed, but that would probably be better handled by removing the checkmark rather than catching a random end of a HTML tag)

Reply:
Tag stripping is already done by datatables. In our case sometimes we want to leave only the inner text of a tag and in other cases we want to remove them entirely(like with the checkmarks).
By using the render function we have more fine granular control, for instance we can strip the checkmarks only in the pdf and exel export and leave them in print and copy.

Reply:
It does but it's broken - it breaks because the `data-content` attribute contains HTML inside (and this is why [you don't parse HTML with regex](https://stackoverflow.com/questions/1732348/regex-match-open-tags-except-xhtml-self-contained-tags/1732454#1732454), or with .split() :P)

I don't need that level of granular control. That's just more code to maintain every time we want to change something in the tables. Removing the checkmarks is the only useful case I can think of (and even that is a workaround just because we can't get it to work)

Reply:
The checkmarks work fine when copying data and printing them. After we fix the checkmarks in the pdf export we can only strip them in the excel export.